File: api-chi/cmd/routes/movie.go
package routes

import (
	"api-chi/cmd/controllers"

	"github.com/go-chi/chi/v5"
)

func MovieRoutes(r chi.Router) {
	controller := controllers.MovieController{}

	r.Route("/movie", func(r chi.Router) {
		r.Get("/count", controller.Count)
		r.Get("/", controller.GetAll)
		r.Post("/", controller.Create)
		r.Patch("/", controller.Update)
		r.Delete("/{id}", controller.Remove)
	})
}

File: api-fastify/src/api/routes/movie.ts
import { FastifyInstance } from "fastify"
import MovieController from "@/api/controllers/movie"

export default async function MovieDirectorRoutes(fastify: FastifyInstance) {
  const controller: MovieController = new MovieController()

  fastify.post("/movies", controller.Create.bind(controller))
  fastify.patch("/movies", controller.Update.bind(controller))
  fastify.delete("/movies/:id", controller.Remove.bind(controller))
  fastify.get("/movies/count", controller.Count.bind(controller))
  fastify.get("/movies", controller.GetAll.bind(controller))
}
